logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/AppendInFlight.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.logstreams.impl.flowcontrol;
+
+import com.netflix.concurrency.limits.Limiter;
+import io.camunda.zeebe.logstreams.storage.LogStorage.AppendListener;
+import io.prometheus.client.Histogram;
+
+/**
+ * Represents an in-flight append. Updates metrics and backpressure limits after being {@link
+ * AppendInFlight#start(long) started} and handles callbacks from the log storage. Commit and write
+ * errors are propagated to an {@link AppendErrorHandler}.
+ */
+public final class AppendInFlight implements AppendListener {
```

Comment:
:upside_down_face: Considering in-flight is a qualifier, I'd call it `InFlightAppend`. Otherwise it sounds like we're appending something called in-flight? Anyway, super nitpick, feel free to ignore :smile:

Reply:
Haha yeah, that's what I had first but then I was annoyed by the alphabetical ordering :D